Sample/app/src/main/java/com/microsoft/projectoxford/visionsample/HandwritingRecognizeActivity.java

```diff
@@ -0,0 +1,240 @@
+        mEditText.setText("");
+
+        Intent intent;
+        intent = new Intent(HandwritingRecognizeActivity.this, com.microsoft.projectoxford.visionsample.helper.SelectImageActivity.class);
+        startActivityForResult(intent, REQUEST_SELECT_IMAGE);
+    }
+
+    // Called when image selection is done.
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Log.d("AnalyzeActivity", "onActivityResult");
+        switch (requestCode) {
+            case REQUEST_SELECT_IMAGE:
+                if (resultCode == RESULT_OK) {
+                    // If image is selected successfully, set the image URI and bitmap.
+                    mImageUri = data.getData();
+
+                    mBitmap = ImageHelper.loadSizeLimitedBitmapFromUri(
+                            mImageUri, getContentResolver());
+                    if (mBitmap != null) {
+                        // Show the image on screen.
+                        ImageView imageView = (ImageView) findViewById(R.id.selectedImage);
+                        imageView.setImageBitmap(mBitmap);
+
+                        // Add detection log.
+                        Log.d("AnalyzeActivity", "Image: " + mImageUri + " resized to " + mBitmap.getWidth()
+                                + "x" + mBitmap.getHeight());
+
+                        doRecognize();
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+
+    public void doRecognize() {
+        mButtonSelectImage.setEnabled(false);
+        mEditText.setText("Analyzing...");
+
+        try {
+            new doRequest().execute();
+        } catch (Exception e) {
+            mEditText.setText("Error encountered. Exception is: " + e.toString());
+        }
+    }
+
+    private String process() throws VisionServiceException, IOException, InterruptedException {
+        Gson gson = new Gson();
+
+        // Put the image into an input stream for detection.
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        mBitmap.compress(Bitmap.CompressFormat.JPEG, 100, output);
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(output.toByteArray());
+
+        //post image and got operation from API
+        HandwritingOCROperation operation = this.client.CreateHandwritingOCROperationAsync(inputStream);
+
+        HandwritingOCROperationResult operationResult;
+        //try to get recognition result until it finished.
+        do {
+            Thread.sleep(1000);
+            operationResult = this.client.GetHandwritingOCROperationResultAsync(operation.Url());
+        }
+        while (operationResult.status.equals("NotStarted") || operationResult.status.equals("Running"));
+        String result = gson.toJson(operationResult);
+        Log.d("result", result);
+
+        return result;
+    }
+
+    private class doRequest extends AsyncTask<String, String, String> {
```

Comment:
This usage may cause memory leak. 
Inner class inside an activity can maintain a reference to this class. If it continues to perform background work after the Activity has been destroyed, the reference to the Activity will persist and it won’t be garbage collected until after the background task completes.